src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -2180,7 +2180,7 @@ private static bool IsEmptyRewritePossible(BoundNode node)
             }
         }
 
-        private sealed class EmptyRewriter : BoundTreeRewriterWithStackGuard
+        private sealed class EmptyRewriter : BoundTreeRewriterWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
```

Comment:
This visitor and its usage were added in #65398. It looks there was no specific motivation around verifying stack guard behavior. Rather it looks like it was about asserting that ```Update``` implementations are working for all possible nodes. The purpose is still preserved, in my opinion.

Reply:
By the way, the change made to ```BoundTreeRewriterWithStackGuard``` around binary patterns probably makes sense since they are used in expression context. However, it is probably not that useful because patterns do not survive the very first rewrite phase (lowering).